btree/btree.c

```diff
@@ -7099,8 +7102,15 @@ static int64_t btree_del_kv_tick(struct m0_sm_op *smop)
 		}
 
 		lev = &oi->i_level[oi->i_used];
+
 		node_slot.s_node = lev->l_node;
 		node_slot.s_idx  = lev->l_idx;
+		node_slot.s_rec  = REC_INIT(&p_key, &ksize, &p_val, &vsize);
+
+		node_rec(&node_slot);
+		node_slot.s_rec.r_flags = M0_BSC_SUCCESS;
+		rc = bop->bo_cb.c_act(&bop->bo_cb, &node_slot.s_rec);
+		M0_ASSERT(rc == 0);
```

Comment:
This should not assert. Instead if the return value is non-0 then abort the delete operation and return back to the caller with the same error code as is added in rc by the callback routine.

Reply:
okay , updated.